### ISSUE TITLE
SWC doesn't reload the application

### ISSUE THREAD

**Issue by afaq1182** [nestjs/nest#11990]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

The swc compiles the file that are changed but doesn't restart the server. I followed exact tutorial as in nestjs official documentation. The files are compiled but the changes aren't reflected in runtime, If I change port in main.ts file and save the file, the app continues running on the previous port.
this is my package.json 
`{
  "name": "total-cx",
  "version": "0.0.1",
  "description": "",
  "author": "",
  "private": true,
  "license": "UNLICENSED",
  "scripts": {
    "build": "nest build",
    "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
    "start": "nest start",
    "start:dev": "nest start --watch",
    "start:swc": "npx nest start -b swc --watch --type-check",
    "start:debug": "nest start --debug --watch",
    "start:prod": "node dist/main",
    "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
    "test": "jest",
    "test:watch": "jest --watch",
    "test:cov": "jest --coverage",
    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
    "test:e2e": "jest --config ./test/jest-e2e.json"
  },
  "dependencies": {
    "@nestjs/common": "^10.0.0",
    "@nestjs/config": "^3.0.0",
    "@nestjs/core": "^10.0.0",
    "@nestjs/mapped-types": "*",
    "@nestjs/platform-express": "^10.0.0",
    "@nestjs/schedule": "^3.0.1",
    "@nestjs/typeorm": "^10.0.0",
    "mysql2": "^3.5.0",
    "reflect-metadata": "^0.1.13",
    "rxjs": "^7.8.1",
    "typeorm": "^0.3.17"
  },
  "devDependencies": {
    "@nestjs/cli": "^10.0.0",
    "@nestjs/schematics": "^10.0.0",
    "@nestjs/testing": "^10.0.0",
    "@swc/cli": "^0.1.62",
    "@swc/core": "^1.3.68",
    "@types/cron": "^2.0.1",
    "@types/express": "^4.17.17",
    "@types/jest": "^29.5.2",
    "@types/node": "^20.3.1",
    "@types/supertest": "^2.0.12",
    "@typescript-eslint/eslint-plugin": "^5.59.11",
    "@typescript-eslint/parser": "^5.59.11",
    "eslint": "^8.42.0",
    "eslint-config-prettier": "^8.8.0",
    "eslint-plugin-prettier": "^4.2.1",
    "jest": "^29.5.0",
    "prettier": "^2.8.8",
    "source-map-support": "^0.5.21",
    "supertest": "^6.3.3",
    "ts-jest": "^29.1.0",
    "ts-loader": "^9.4.3",
    "ts-node": "^10.9.1",
    "tsconfig-paths": "^4.2.0",
    "typescript": "^5.1.3"
  },
  "jest": {
    "moduleFileExtensions": [
      "js",
      "json",
      "ts"
    ],
    "rootDir": "src",
    "testRegex": ".*\\.spec\\.ts$",
    "transform": {
      "^.+\\.(t|j)s$": "ts-jest"
    },
    "collectCoverageFrom": [
      "**/*.(t|j)s"
    ],
    "coverageDirectory": "../coverage",
    "testEnvironment": "node"
  }
}
`

and this is `.swcrc` 
`{
    "sourceMaps": true,
    "module": {
      "type": "commonjs",
      "strict": true
    },
    "jsc": {
      "target": "es2021",
      "parser": {
        "syntax": "typescript",
        "decorators": true
      },
      "transform": {
        "legacyDecorator": true,
        "decoratorMetadata": true
      },
      "keepClassNames": true
    }
  }`

commands I have tried 
`nest start -b swc -w --type-check`,
`npx nest start -b swc --watch --type-check`,
this is `nest-cli.json`,
`{
  "$schema": "https://json.schemastore.org/nest-cli",
  "collection": "@nestjs/schematics",
  "sourceRoot": "src",
  "compilerOptions": {
    "deleteOutDir": true,
    "builder": "swc",
    "typeCheck": true
  }
}
`

### Minimum reproduction code

https://stackblitz.com/edit/nestjs-typescript-starter-extwt2?file=nest-cli.json,.swcrc,src%2Fapp.service.ts,package.json

### Steps to reproduce

normal steps

### Expected behavior

I saw on Kamil twitter in which when he saves the file, the server restarts immediately.

### Package

- [ ] I don't know. Or some 3rd-party package
- [ ] <code>@nestjs/common</code>
- [ ] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [X] Other (see below)

### Other package

swc

### NestJS version

10

### Packages versions

```json

```

### Node.js version

19.x

### In which operating systems have you tested?

- [ ] macOS
- [X] Windows
- [ ] Linux

**Comment by micalevisk** [comment]
seems to be working with the latest version of `@nestjs/cli` (v10.1.7)

try that out.

**Comment by huthaifaa** [comment]
> it seems to be working in the latest version of `@nestjs/cli` (v10.1.7)


Working well so far, thanks @micalevisk

**Comment by JohanManders** [comment]
Could you check if this is really resolved @micalevisk?

I had the same problem with my project so I downloaded @afaq1182 his/her stackblitz repo to check if it is the same problem.

I changed the ```package.json``` file so that ```@nestjs/cli``` would use version ```10.1.8```.

When running ```nest start --builder swc --watch``` and change the return string inside ```app.service.ts``` I do get ```Successfully compiled src\app.service.ts with swc (3.91ms)``` in the terminal and the file in the ```dist``` folder is updated, but nest does not restart, so visiting that page again gives the old string.

This makes debugging not easy with SWC.

I am using Windows 10 and Node v18.13.0.

**Comment by micalevisk** [comment]
@JohanManders it worked here on Linux node v18

**Comment by JohanManders** [comment]
@micalevisk Windows might be the problem then. 

Version ```10.0.5``` of ```@nestjs/cli``` works fine, but every version after that does not.

I see that there is a similar closed issue in the cli repo (https://github.com/nestjs/nest-cli/issues/2152) that states that version ```10.1.2``` should have fixed it, but it did not for me.

**Comment by hosup77** [comment]
> @micalevisk Windows might be the problem then.
> 
> Version `10.0.5` of `@nestjs/cli` works fine, but every version after that does not.
> 
> I see that there is a similar closed issue in the cli repo ([nestjs/nest-cli#2152](https://github.com/nestjs/nest-cli/issues/2152)) that states that version `10.1.2` should have fixed it, but it did not for me.

Same issue using Window 11, node 20, @nestjs/cli `10.1.9`.

**Comment by micalevisk** [comment]
@hosup77 or @JohanManders 

does it works when using `@nestjs/cli@10.1.6`?

**Comment by micalevisk** [comment]
so there's something wrong hee: https://github.com/nestjs/nest-cli/releases/tag/10.1.0

**Comment by micalevisk** [comment]
I believe that that import from `path/posix` was intentional but I don't know the reasons. Here: https://github.com/nestjs/nest-cli/commit/2d0c6fd3938d7c865eb2137ecd53634b71845fa3

but feel free to open a PR at https://github.com/nestjs/nest-cli to propose that change

**Comment by kamilmysliwiec** [comment]
Yeah that's odd indeed - `options.cliOptions.outDir` should already be converted from win to posix thanks to `convertPath` call
